### ISSUE TITLE
Figure out how to handle masking for RoBERTa

### ISSUE THREAD

**Issue by matt-gardner** [allenai/allennlp#3359]
For some reason, the RoBERTa folks chose to use 0 for the [CLS] token instead of for padding, which breaks a bunch of assumptions that we make in our code.  We need to figure out a way to get a mask for RoBERTa that doesn't require the model knowing whether it's using BERT or RoBERTa.  I'm not really sure how to do this, other than to add a key to the token indexer output dictionary, or something (which I believe is also what some of the more complex indexers do already).

**Comment by ChunchuanLv** [comment]
We can pre-decide all the special token ids in constructor, then them in forward pass, instead of using 0 or 1 directly.

For example, in token embedder init, we have

```
        if "roberta" in model_name:
            self.transformer_model = RobertaModel.from_pretrained(model_name, output_hidden_states=True)
            self.pad_id = 1  #or call some function
        else:
            self.transformer_model = BertModel.from_pretrained(model_name, output_hidden_states=True)
            self.pad_id = 0
```
Then,  in forward

```
        input_mask = (input_ids != self.pad_id).long()
```

Similarly for the token indexer. Maybe, I missed something?

**Comment by matt-gardner** [comment]
This is the method that we use to get the mask: https://github.com/allenai/allennlp/blob/master/allennlp/nn/util.py#L555-L602.  The issue isn't just when computing vectors, it's also when normalizing attentions, computing a loss over non-padded word indices, etc.  We need the mask outside of the `TokenEmbedder`.

**Comment by matt-gardner** [comment]
I believe this is fixed by #3560.  If someone thinks it's not, please comment here or open a new issue.
